### PR TITLE
Added support for Apache HTTP Client ConnectionKeepAliveStrategy and ConnectionReuseStrategy

### PR DESCRIPTION
**Pull request by rsandidfc**
Added support for Apache HTTP Client ConnectionKeepAliveStrategy and ConnectionReuseStrategy

3rd time's the charm? Added the support, checkstyle conforms, copywrite years updated, and signed-off-by is below. Fingers crossed..........

Signed-off-by: Richard A Sand <rsand@idfconnect.com>

### CODE REVIEW

File: connectors/apache-connector/src/main/java/org/glassfish/jersey/apache/connector/ApacheConnector.java [hunk 1]
```diff
@@ -197,6 +200,32 @@
             }
         }
 
+        final Object keepAliveStrategy = config.getProperties().get(ApacheClientProperties.KEEPALIVE_STRATEGY);
+        if (keepAliveStrategy != null) {
+            if (!(keepAliveStrategy instanceof ConnectionKeepAliveStrategy)) {
+                LOGGER.log(
+                        Level.WARNING,
+                        LocalizationMessages.IGNORING_VALUE_OF_PROPERTY(
```
Comment (senivam) [inline connectors/apache-connector/src/main/java/org/glassfish/jersey/apache/connector/ApacheConnector.java]: the code actually does not ignore value of the property, it only logs intention to do so. And the if inside another if could be rewritten as if (A && !B).

File: connectors/apache-connector/src/main/java/org/glassfish/jersey/apache/connector/ApacheConnector.java [hunk 2]
```diff
@@ -197,6 +200,32 @@
...
+                );
+            }
+        }
+
+        final Object reuseStrategy = config.getProperties().get(ApacheClientProperties.REUSE_STRATEGY);
+        if (reuseStrategy != null) {
+            if (!(reuseStrategy instanceof ConnectionReuseStrategy)) {
+                LOGGER.log(
+                        Level.WARNING,
+                        LocalizationMessages.IGNORING_VALUE_OF_PROPERTY(
```
Comment (senivam) [inline connectors/apache-connector/src/main/java/org/glassfish/jersey/apache/connector/ApacheConnector.java]: The same thing is valid here: 
the code actually does not ignore value of the property, it only logs intention to do so. And the if inside another if could be rewritten as if (A && !B).

Comment (rsandidfc) [reply connectors/apache-connector/src/main/java/org/glassfish/jersey/apache/connector/ApacheConnector.java]: Whoops, how silly, I just copied the connectionManager block and didn't change it to null the values if invalid. I'll fix now

Comment (rsandidfc) [reply connectors/apache-connector/src/main/java/org/glassfish/jersey/apache/connector/ApacheConnector.java]: Ok I've committed the update to the pull

**Pull request merged**